init/libfido2-util.go

```diff
@@ -0,0 +1,330 @@
+package main
+
+/*
+#cgo CFLAGS: -I/usr/include/
+#cgo LDFLAGS: -lfido2
+#include <stdio.h>
+#include <stdlib.h>
+#include <fido.h>
+*/
+import "C"
+import (
+	"fmt"
+	"sync"
+	"unsafe"
+)
+
+const (
+	Default OptionValue = ""
+	True    OptionValue = "true"
+	False   OptionValue = "false"
+)
+
+const (
+	HMACSecretExtension  Extension = "hmac-secret"
+	CredProtectExtension Extension = "credProtect"
+)
+
+const (
+	// ErrInvalidArgument if arguments are invalid.
+	ErrInvalidArgument = "invalid argument"
```

Comment:
you can make them error type directly: `ErrInvalidArgument = fmt.Errorf("invalid argument")`. And then have a static map of `C error code -> go error` that will look more elegant.